testsuite/forge/src/backend/k8s/fullnode.rs

```diff
@@ -0,0 +1,738 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    Create, Get, K8sNode, ReadWrite, Result, Version, REST_API_SERVICE_PORT,
+    VALIDATOR_0_DATA_PERSISTENT_VOLUME_CLAIM_PREFIX, VALIDATOR_0_GENESIS_SECRET_PREFIX,
+    VALIDATOR_0_STATEFUL_SET_NAME,
+};
+use anyhow::Context;
+use aptos_config::{
+    config::{
+        merge_node_config, BaseConfig, DiscoveryMethod, ExecutionConfig, NetworkConfig, NodeConfig,
+        RoleType, WaypointConfig,
+    },
+    network_id::NetworkId,
+};
+use aptos_logger::info;
+use aptos_sdk::types::PeerId;
+use aptos_short_hex_str::AsShortHexStr;
+use k8s_openapi::{
+    api::{
+        apps::v1::{StatefulSet, StatefulSetSpec},
+        core::v1::{
+            ConfigMap, ConfigMapVolumeSource, Container, PersistentVolumeClaim,
+            PersistentVolumeClaimSpec, PodSpec, PodTemplateSpec, ResourceRequirements,
+            SecretVolumeSource, Service, ServicePort, ServiceSpec, Volume, VolumeMount,
+        },
+    },
+    apimachinery::pkg::apis::meta::v1::LabelSelector,
+};
+use kube::api::{ObjectMeta, PostParams};
+use std::{collections::BTreeMap, path::PathBuf, sync::Arc};
+use tempfile::TempDir;
+
+const FULLNODE_CONFIG_MAP_KEY: &str = "fullnode.yaml";
+
+// these are constants given by the aptos-node helm chart
+// see terraform/helm/aptos-node/templates/validator.yaml
+
+// the path where the genesis is mounted in the validator
+const GENESIS_CONFIG_VOLUME_NAME: &str = "genesis-config";
+const GENESIS_CONFIG_VOLUME_PATH: &str = "/opt/aptos/genesis";
+
+// the path where the config file is mounted in the fullnode
+const APTOS_CONFIG_VOLUME_NAME: &str = "aptos-config";
+const APTOS_CONFIG_VOLUME_PATH: &str = "/opt/aptos/etc";
+
+// the path where the data volume is mounted in the fullnode
+const APTOS_DATA_VOLUME_NAME: &str = "aptos-data";
+const APTOS_DATA_VOLUME_PATH: &str = "/opt/aptos/data";
+
+/// Get the image repo from the validator. The fullnode will use the same image repo, as they both use the "validator" image
+fn get_image_repo_from_validator(validator_stateful_set: &StatefulSet) -> Result<String> {
```

Comment:
Can you reuse the `get_stateful_set_image` function instead?

Reply:
ah you're right! I forgot about that function... 🤦